packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,195 @@
+      width: `${(category.value / options.maximumValue) * 100}%`
+    };
+
+    const cssClasses = {
+      'as-category-widget__category': true,
+      'as-category-widget__category--not-selected': isAnyCategorySelected && (!isSelected || options.isOther),
+      'as-category-widget__category--other': options.isOther,
+      'as-category-widget__category--selected': isSelected
+    };
+
+    return (
+      <li class={cssClasses} onClick={() => this._toggleCategory(category)}>
+        <p class='as-category-widget__category-title as-body' data-value={readableNumber(category.value)}>
+          {category.name}
+        </p>
+
+        <div class='as-category-widget__bar'>
+          <div class='as-category-widget__value' style={progressStyles}></div>
+        </div>
+      </li>
+    );
+  }
+
+  private _renderOtherCategory(category: Category, options: CategoryOptions) {
+    const categoryData = category || {
+      name: 'Other',
+      value: this._getCategoriesTotalValue(
+        this.categories.slice(this.numberOfVisibleCategories, this.categories.length)
+      )
+    };
+
+    return this._renderCategory(categoryData, { maximumValue: options.maximumValue, isOther: true });
+  }
+
+  private _renderFooter() {
+    const selectedCount = this.selectedCategories.length ? this.selectedCategories.length : 'All';
+
+    return (
+      <footer class='as-category-widget__footer'>
+        <div class='as-category-widget__count as-body'>{selectedCount} selected</div>
+        { this.showClearButton ? <button class='' onClick={() => this.clearSelection()}>Clear selection</button> : '' }
+      </footer>
+    );
+  }
+
+  private _isSelected(categoryName: string) {
+    return this.selectedCategories.indexOf(categoryName) > -1;
+  }
+
+  private _toggleCategory(category: Category) {
+    const newSelectedCategories = this._isSelected(category.name)
+      ? this.selectedCategories.filter((currentCategory) => currentCategory !== category.name)
+      : [...this.selectedCategories, category.name];
+
+    this.selectedCategories = newSelectedCategories;
+    this._onCategoriesChanged();
+  }
+
+  private _onCategoriesChanged() {
+    this.categoriesSelected.emit(this.selectedCategories);
+  }
+
+  private _getVisibleCategoriesMaximumValue() {
+    const categories = this.categories.slice(0, this.numberOfVisibleCategories);
+
+    return categories.reduce(
+      (maximum, currentCategory: Category) => currentCategory.value > maximum ? currentCategory.value : maximum, 0
+    );
+  }
+
+  private _getCategoriesTotalValue(categories: object[]) {
+    return categories.reduce(
+      (maximum, currentCategory: Category) => currentCategory.value + maximum, 0
+    );
+  }
+
+  private _getBarColor(color: string, options: { isSelected?: boolean } = {}) {
+    if (options.isSelected) {
+      return shadeOrBlend(-0.16, color);
+    }
+
+    return color;
+  }
+
+  private parseCategories() {
+    const newCategories = [...this.categories];
```

Comment:
do we need the spread operator here?

Reply:
Yes, we need it because we're cloning the array, `[this.categories]` wouldn't work.